include/hnswlib/space_ip.h

```diff
@@ -3,289 +3,379 @@
 
 namespace hnswlib {
 
-static float InnerProduct(const void *pVect1, const void *pVect2,
-                          const void *qty_ptr,
-                          const local_state_t *local_state) {
-  size_t qty = *((size_t *)qty_ptr);
-  float res = 0;
-  for (unsigned i = 0; i < qty; i++) {
-    res += ((float *)pVect1)[i] * ((float *)pVect2)[i];
-  }
-  return (1.0f - res);
-}
+    static float
+    InnerProduct(const void *pVect1, const void *pVect2, const void *qty_ptr, const local_state_t *local_state) {
+        size_t qty = *((size_t *) qty_ptr);
+        float res = 0;
+        for (unsigned i = 0; i < qty; i++) {
+            res += ((float *) pVect1)[i] * ((float *) pVect2)[i];
+        }
+        return res;
+
+    }
+
+    static float
+    InnerProductDistance(const void *pVect1, const void *pVect2, const void *qty_ptr, const local_state_t *local_state) {
+        return 1.0f - InnerProduct(pVect1, pVect2, qty_ptr, local_state);
+    }
```

Comment:
In `space_ip.h` and `space_l2.h`, some new functions added. I add the parameters like `const local_state_t *local_state` to these new functioins like others to make the parameter lists non-conflicting.